src/osdc/ObjectCacher.cc

```diff
@@ -414,7 +420,7 @@ ObjectCacher::BufferHead *ObjectCacher::Object::map_write(ObjectExtent &ex,
     // at end ?
     if (p == data.end()) {
       if (final == NULL) {
-        final = new BufferHead(this);
+        final = new BufferHead(this, trace);
```

Comment:
Look at the cases where ```replace_journal_tid``` is invoked -- when you are overwriting an extent, I think you should replace the trace